packages/manager/apps/web-hosting/src/pages/dashboard/ssl/TopBar.component.tsx

```diff
@@ -1,25 +1,53 @@
 import React, { useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
 
-import { OdsButton } from '@ovhcloud/ods-components/react';
+import { OdsButton, OdsSelect, OdsText } from '@ovhcloud/ods-components/react';
 import { useResourcesIcebergV2 } from '@ovh-ux/manager-react-components';
-import { ODS_BUTTON_SIZE, ODS_BUTTON_VARIANT } from '@ovhcloud/ods-components';
+import {
+  ODS_BUTTON_SIZE,
+  ODS_BUTTON_VARIANT,
+  ODS_TEXT_PRESET,
+} from '@ovhcloud/ods-components';
+import { AttachedDomain } from '@/types/ssl';
+import { useCreateDomainCertificate } from '@/data/hooks/useSsl';
 import SectigoModal from './modals/SectigoModal';
 import ImportModal from './modals/ImportModal';
 
 type TTopbar = {
   setMessage: ({ status, label }: { status: string; label: string }) => void;
 };
 export default function Topbar({ setMessage }: Readonly<TTopbar>) {
+  const { serviceName } = useParams();
   const [hasSectigoModal, setHasSectigoModal] = useState<boolean>(false);
   const [hasImportModal, setHasImportModal] = useState<boolean>(false);
+  const [selectedDomains, setSelectedDomains] = useState<string[]>([]);
 
   const { t } = useTranslation('ssl');
   const { flattenData } = useResourcesIcebergV2({
     route: `/webhosting/attachedDomain`,
-    queryKey: [],
+    queryKey: ['attached-domain'],
   });
 
+  const { createDomainCertificate } = useCreateDomainCertificate(
+    serviceName,
+    () => {
+      setMessage({
+        status: 'success',
+        label: t('hosting_dashboard_ssl_order_success'),
+      });
+    },
+    () => {
+      setMessage({
+        status: 'error',
+        label: t('hosting_dashboard_ssl_order_error'),
+      });
+    },
+  );
+  const handleSslEncrypt = () => {
+    selectedDomains?.map((domain) => createDomainCertificate(domain));
```

Comment:
Maybe you should make a `createDomainCertificates` function that takes an array of domain and use `Promise.all` or `Promise.allSettled` to send them, that way onSuccess and onError will be called only once after all actions are performed, and not for each action